source/common/http/http2/codec_impl.cc

```diff
@@ -90,7 +90,19 @@ void ConnectionImpl::StreamImpl::encode100ContinueHeaders(const HeaderMap& heade
 
 void ConnectionImpl::StreamImpl::encodeHeaders(const HeaderMap& headers, bool end_stream) {
   std::vector<nghttp2_nv> final_headers;
-  buildHeaders(final_headers, headers);
+
+  if (Http::Utility::isUpgrade(headers)) {
+    HeaderMapImpl modified_headers(headers);
+    upgrade_type_ = headers.Upgrade()->value().c_str();
+    if (headers.Status()) {
+      Http::Utility::transformUpgradeResponseFromH1toH2(modified_headers);
```

Comment:
My high level comment above the encode/decode changes in this file is that it seems to assume that above the codec level, all upgrades are treated as H1 style upgrades. Basically, this is an invariant that we assume is true for both upstream and downstream proxy. Is that correct? If so, I think it makes sense to do the conversions in the codec, but perhaps more comments on what is going on?

Reply:
shouldn't this be transformUpgradeResponseFromH2toH1?

Reply:
ignore..